Sources/ClientRuntime/Middleware/RetryMiddleware.swift

```diff
@@ -66,3 +102,24 @@ public struct RetryMiddleware<Strategy: RetryStrategy,
         }
     }
 }
+
+// Calculates & returns TTL datetime in strftime format `YYYYmmddTHHMMSSZ`.
+func getTTL(now: Date, estimatedSkew: TimeInterval, socketTimeout: TimeInterval) -> String {
+    let dateFormatter = DateFormatter()
+    dateFormatter.dateFormat = "yyyyMMdd'T'HHmmss'Z'"
+    dateFormatter.locale = Locale(identifier: "en_US_POSIX")
+    dateFormatter.timeZone = TimeZone(abbreviation: "GMT")
+    let ttlDate = now.addingTimeInterval(estimatedSkew + socketTimeout)
+    return dateFormatter.string(from: ttlDate)
+}
```

Comment:
Utility method for adding `socketTimeout` and `estimatedSkew` to given `Date`. Returns the sum as a string in strftime format.